File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.PersonService.Models/isave-person_result.ts



import { PersonLanguage_Item } from './person-language_item';

export interface ISavePerson_Result {
  
	PersonID?: number;
	FirstMiddleNames: string;
	LastNames: string;
	Gender: string;
	UnitID?: number;
	IsUSCitizen: boolean;
	NationalID: string;
	ResidenceStreetAddress: string;
	ResidenceCityID?: number;
	ContactEmail: string;
	ContactPhone: string;
	DOB?: Date;
	POBCityID?: number;
	FatherName: string;
	MotherName: string;
	HighestEducationID?: number;
	FamilyIncome?: number;
	EnglishLanguageProficiencyID?: number;
	PassportNumber: string;
	PassportExpirationDate?: Date;
	PassportIssuingCountryID?: number;
	LawPoliceMilitaryID: string;
	Rank?: number;
	YearsInCurrentPosition?: number;
	MedicalClearanceStatus?: boolean;
	MedicalClearanceDate?: Date;
	DentalClearanceStatus?: boolean;
	DentalClearanceDate?: Date;
	PsychologicalClearanceStatus?: boolean;
	PsychologicalClearanceDate?: Date;
	ModifiedByAppUserID?: number;
	Languages?: PersonLanguage_Item[];

}

File: GTTS/INL.PersonService.Models/ISavePerson_Result.cs
using INL.Services.Models;
using System;
using System.Collections.Generic;

namespace INL.PersonService.Models
{
    public interface ISavePerson_Result : IBaseResult
    {
        long? PersonID { get; set; }
        string FirstMiddleNames { get; set; }
        string LastNames { get; set; }
        char Gender { get; set; }
        long? UnitID { get; set; }
        bool IsUSCitizen { get; set; }
        string NationalID { get; set; }
        string ResidenceStreetAddress { get; set; }
        int? ResidenceCityID { get; set; }
        string ContactEmail { get; set; }
        string ContactPhone { get; set; }
        DateTime? DOB { get; set; }
        int? POBCityID { get; set; }
        string FatherName { get; set; }
        string MotherName { get; set; }
        int? HighestEducationID { get; set; }
        decimal? FamilyIncome { get; set; }
        int? EnglishLanguageProficiencyID { get; set; }
        string PassportNumber { get; set; }
        DateTime? PassportExpirationDate { get; set; }
        int? PassportIssuingCountryID { get; set; }
        string LawPoliceMilitaryID { get; set; }
        int? Rank { get; set; }
        int? YearsInCurrentPosition { get; set; }
        bool? MedicalClearanceStatus { get; set; }
        DateTime? MedicalClearanceDate { get; set; }
        bool? DentalClearanceStatus { get; set; }
        DateTime? DentalClearanceDate { get; set; }
        bool? PsychologicalClearanceStatus { get; set; }
        DateTime? PsychologicalClearanceDate { get; set; }
        int? ModifiedByAppUserID { get; set; }

        List<PersonLanguage_Item> Languages { get; set; }
    }
}
